app/controllers/DepartureTestMessagesController.scala

```diff
@@ -52,47 +55,46 @@ class DepartureTestMessagesController @Inject()(cc: ControllerComponents,
-                      postResponse =>
-                        postResponse.status match {
-                          case status if is2xx(status) =>
-                            postResponse.header(LOCATION) match {
-                              case Some(locationValue) =>
-                                val messageId = Utils.lastFragment(locationValue)
-                                departureMessageConnector.get(departureId.index.toString, messageId).map {
-                                  case Right(movementMessage: MovementMessage) =>
-                                    Created(
-                                      HateaosDepartureResponse(
-                                        departureId,
-                                        request.testMessage.messageType,
-                                        request.generatedMessage,
-                                        locationValue
-                                      )
-                                    )
-                                  case Left(getMessageResponse) =>
-                                    handleNon2xx(getMessageResponse)
-                                }
-                              case None =>
-                                Future.successful(InternalServerError)
+            case Right(departureWithMessages: DepartureWithMessages) =>
+              val generatedMessage =
+                XMLTransformer.populateRefNumEPT1(request.generatedMessage, MessageType.DepartureDeclaration.code, departureWithMessages.messages)
+              inboundRouterConnector
+                .post(request.testMessage.messageType, generatedMessage.toString(), departureId.index)
+                .flatMap {
+                  postResponse =>
+                    postResponse.status match {
+                      case status if is2xx(status) =>
+                        postResponse.header(LOCATION) match {
+                          case Some(locationValue) =>
+                            val messageId = Utils.lastFragment(locationValue)
+                            departureMessageConnector.get(departureId.index.toString, messageId).map {
+                              case Right(_) =>
+                                Created(
+                                  HateaosDepartureResponse(
+                                    departureId,
+                                    request.testMessage.messageType,
+                                    generatedMessage,
+                                    locationValue
+                                  )
+                                )
+                              case Left(getMessageResponse) =>
+                                handleNon2xx(getMessageResponse)
                             }
-                          case _ =>
-                            Future.successful(handleNon2xx(postResponse))
+                          case None =>
+                            Future.successful(InternalServerError)
```

Comment:
This will happen as handleNon2xx is used when getMessages returns a Left:
https://github.com/hmrc/common-transit-convention-traders-test-support/blob/0701e2ac7af9a88bcc10356f21b71b86e32237de/app/controllers/DepartureTestMessagesController.scala#L97

This case block returns 500 Internal Server Error when there is no Location header in the response from the inbound router.